server/routes.go

```diff
@@ -121,8 +122,8 @@ func (s *Server) GenerateHandler(c *gin.Context) {
 		return
 	}
 
-	if model.IsEmbedding() {
-		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "embedding models do not support generate"})
+	if !model.Has(CapabilityCompletion) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "model does not support completion"})
```

Comment:
This error should be a little more user friendly since people are accessing it via:

* `ollama run`
* `/api/chat`
* `/api/generate`

and may not understand what "completion" is.